src/main/java/org/reaktivity/nukleus/socks/internal/stream/SocksServerFactory.java

```diff
@@ -0,0 +1,801 @@
+            }
+            else if (payload != null)
+            {
+                decodeTraceId = data.trace();
+                DirectBuffer buffer = payload.buffer();
+                int offset = payload.offset();
+                int limit = payload.limit();
+                this.authorization = data.authorization();
+                this.decodeTraceId = data.typeId();
+
+                if (bufferSlot != BufferPool.NO_SLOT)
+                {
+                    MutableDirectBuffer decodeBuffer = bufferPool.buffer(bufferSlot);
+                    decodeBuffer.putBytes(bufferSlotOffset, buffer, offset, limit - offset);
+                    bufferSlotOffset += limit - offset;
+                    buffer = decodeBuffer;
+                    offset = 0;
+                    limit = bufferSlot;
+                }
+
+                while(offset < limit)
+                {
+                    offset += decodeState.decode(buffer, offset, limit);
+                }
+            }
+            else
+            {
+                //TODO
+            }
+
+        }
+
+        private void onNetworkBegin(
+            BeginFW begin)
+        {
+            doNetworkBegin(supplyTraceId.getAsLong());
+        }
+
+        private void onNetworkEnd(
+            EndFW end)
+        {
+            final long traceId = end.trace();
+            doNetworkEnd(traceId);
+        }
+
+        private void onNetworkAbort(
+            AbortFW abort)
+        {
+            final long traceId = abort.trace();
+            doNetworkAbort(traceId);
+        }
+
+        private void onNetworkWindow(
+            WindowFW window)
+        {
+            final int replyCredit = window.credit();
+
+            replyBudget += replyCredit;
+            replyPadding += window.padding();
+
+            final int initialCredit = bufferPool.slotCapacity() - initialBudget;
+            doNetworkWindow(supplyTraceId.getAsLong(), initialCredit);
+        }
+
+        private void onSocksCommandRequest(
+            SocksCommandRequestFW socksCommandRequest)
+        {
+            if (socksCommandRequest.version() != 5 && socksCommandRequest.reserved() != 0)
+            {
+                //TODO
+            }
+
+            switch (socksCommandRequest.command().get())
+            {
+                case CONNECT:
+                    onSocksConnect(socksCommandRequest);
+                    break;
+                case BIND:
+                    //TODO
+                    break;
+                case UDP_ASSOCIATE:
+                    //TODO
+                    break;
+            }
+        }
+
+        private void onSocksConnect(
+            SocksCommandRequestFW socksCommandRequest)
+        {
+            final MessagePredicate filter = (t, b, o, l) ->
+            {
+                if (t > 0 && t <= ResetFW.TYPE_ID && o < l)
+                {
+                    return true;
+                }
+                return false;
+            };
```

Comment:
What is the intent here?

Reply:
I don't really know how to write the filter function. I just make sure `offset < length`, do you have any suggestions about how to write the filter function or just return `true` for now?

Reply:
The purpose of the filter is to determine whether or not you have a matching route for the requested `CONNECT` `address` and `port`.

See https://github.com/reaktivity/nukleus-http.java/blob/develop/src/main/java/org/reaktivity/nukleus/http/internal/stream/ServerAcceptStream.java#L1031 as an example.

Reply:
I am still confused about this one because I don't have the `header`. I came up with two approaches:

- Check if I can create route 

- check if the `socksCommandRequest` has valid address and port.

Reply:
The general goal of the route filter is to compare what will become`SocksBeginEx` metadata to each `SocksRouteEx` metadata to find a matching route.

Since the `SocksBeginEx` is only created after a matching route is found, then you need to instead compare `SocksAddress` plus `port` received in `SocksCommandRequest` to `SocksRouteEx` `address` and `port`.

One way to think about this is that the `SocksRouteEx` places a constraint on a `route` that must be fulfilled by the `SocksCommandRequest` in order to follow that `route`.

Therefore, one subtlety is that a `route` with no `SocksRouteEx` on it is unconstrained, meaning that it matches any `SocksCommandRequest` so should also return `true` from the route filter in this case.

That's why the `nukleus-http` example is defaulting to `true` and doing extra work to confirm the match only when the `HttpRouteEx` extension is present on the route.